src/components/ui-custom/SearchBar.tsx

```diff
@@ -1,14 +1,33 @@
 "use client";
 import clsx from "clsx";
 import { Input } from "../ui/input";
+import { useRouter, useSearchParams, usePathname } from "next/navigation";
+import { useDebouncedCallback } from "use-debounce";
 
-// TODO: It will be done in next PR
 const SearchBar = ({ className }: { className?: string }) => {
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const { replace } = useRouter();
+
+  const handleSearch = useDebouncedCallback((searchTerm: string) => {
+    const params = new URLSearchParams(searchParams);
+    if (searchTerm) {
+      params.set("query", searchTerm);
+    } else {
+      params.delete("query");
+    }
+    replace(`${pathname}?${params.toString()}`);
```

Comment:
isnt `toString()` redundant inside template literal?

Reply:
In my opinion no. We don't need const router in the code, because we are using only replace method, and replace in the first place immediately tells what's going on